be/src/exec/sorting/merge_path.cpp

```diff
@@ -235,8 +231,8 @@ void detail::_do_merge_along_merge_path(const SortDescs& descs, const InputSegme
     };
 
     size_t di = start_di;
-    while (di - start_di < length && di < dest.total_len) {
-        const size_t max_step = std::min(length - (di - start_di), dest.total_len - di);
+    while (di - start_di < length) {
+        const size_t max_step = length - (di - start_di);
```

Comment:
would it overflow?

Reply:
do you mean `length` may smaller than `(di - start_di)`?

It is guarantee at 

```diff
- const size_t length = next_start_di - start_di;
+ const size_t length = is_last_parallelism ? (dest.total_len - start_di) : (next_start_di - start_di);
```